src/lib/propagation/node.js

```diff
@@ -0,0 +1,79 @@
+import * as constants from './constants'
+
+const {
+  NODE_TYPES,
+  nodeTypeToGuesstimateType,
+  ERROR_TYPES: {GRAPH_ERROR},
+  ERROR_SUBTYPES: {GRAPH_SUBTYPES: {MISSING_INPUT_ERROR, IN_INFINITE_LOOP, INVALID_ANCESTOR_ERROR}},
+} = constants
+
+import {Guesstimator} from 'lib/guesstimator/index'
+import {_matchingFormatter} from 'lib/guesstimator/formatter/index'
+
+import * as _collections from 'gEngine/collections'
+import * as _utils from 'gEngine/utils'
+
+export class SimulationNode {
+  constructor({id, expression, type, guesstimateType, samples, errors, childrenIndices, parentIndices, ancestors}, DAG, index) {
+    this.id = id
+    this.expression = expression
+    this.type = type
+    this.guesstimateType = guesstimateType
+    this.samples = samples
+    this.errors = errors
+    this.parentIndices = parentIndices
+    this.childrenIndices = childrenIndices
+    this.ancestors = ancestors
+    this.DAG = DAG
+    this.index = index
+  }
+
+  data() { return this.type === NODE_TYPES.DATA ? this.samples : [] }
+
+  parse() {
+    const e = { text: this.expression, guesstimateType: this.guesstimateType, data: this.data() }
+    const formatter = _matchingFormatter(e)
+    return [formatter.error(e), formatter.format(e)]
+  }
+
+  getInputs() {
```

Comment:
I considered it, but didn't think it would make much sense as these are true functions more so than property accessors. Would you prefer I go that route, or was this a suggestion for consideration rather than one demonstrating opinion?